sqlglot/dialects/mysql.py

```diff
@@ -617,6 +613,32 @@ def _parse_chr(self) -> t.Optional[exp.Expression]:
 
             return self.expression(exp.Chr, **kwargs)
 
+        def _parse_group_concat(self) -> t.Optional[exp.Expression]:
+            def concat_exprs(node, exprs):
+                if node and isinstance(node, exp.Distinct) and len(node.expressions) > 1:
+                    concat_exprs = [self.expression(exp.Concat, expressions=node.expressions)]
+                    node.set("expressions", concat_exprs)
+                    return node
+                return (
+                    exprs[0] if len(exprs) == 1 else self.expression(exp.Concat, expressions=args)
+                )
+
+            args = self._parse_csv(self._parse_lambda)
+            order = args[-1] if isinstance(args[-1], exp.Order) else None
```

Comment:
Let's instead do:

```python
order = seq_get(args, -1)
if not isinstance(order, exp.Order):
    order = None
```

Since SQLGlot allows partial parsing (`error_level=ErrorLevel.IGNORE`), we don't want this to raise an index error if there are no args.